library/Mockery/Generator/Parameter.php

```diff
@@ -56,7 +62,13 @@ public function getTypeHintAsString()
         }
 
         if (preg_match('/^Parameter #[0-9]+ \[ \<(required|optional)\> (?<typehint>\S+ )?.*\$' . $this->rfp->getName() . ' .*\]$/', $this->rfp->__toString(), $typehintMatch)) {
+
             if (!empty($typehintMatch['typehint'])) {
+
+                if ($this->isNamespacedClass($typehintMatch['typehint'])) {
```

Comment:
@robertbasic , please proceed with creating/using method that determines if it's an internal type (e.g. array/callable) to avoid adding `\\` before them. Also add tests for such case.